backend/controllers/userController.js

```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+const User = require('../models/userModel.js');
+
+const bcrypt = require('bcrypt');
+const saltRounds = 12; // you can adjust this value as needed
+
+const createUser = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const encryptedPassword = await bcrypt.hash(password, saltRounds);
```

Comment:
Nothing wrong here, mostly just a nitpick to make the destructuring of the req.body params either both inside the try block or outside of it between the login and createUser functions.   Mostly for code consistency between the two functions.